packages/sentry/index.tsx

```diff
@@ -0,0 +1,29 @@
+import * as Sentry from '@sentry/nextjs';
+
+export const sentryFetch: typeof fetch = async (resource, options) => {
+   const context = {
+      // Underscore sorts the resource first in Sentry's UI
+      _resource: resource,
+      headers: options?.headers,
+      method: options?.method,
+      // Parse to pretty print GraphQL queries
+      body: options?.body ? JSON.parse(String(options?.body)) : undefined,
+   };
+   return fetch(resource, options)
+      .then((response) => {
+         if (response.status >= 400 && response.status !== 401) {
+            const msg = `\`fetch()\` HTTP error: ${response.status} ${response.statusText}`;
+            Sentry.captureException(new Error(msg), (scope) => {
+               scope.setContext('request', context);
+               return scope;
+            });
+            console.error(msg, context);
+         }
+         return response;
+      })
+      .catch((error) => {
+         // We don't want to hear about network errors in Sentry
```

Comment:
I think you're probably right, but this code doesn't prevent them from getting to sentry. They may still be unhandled promise rejections, but I suppose we should handle those at each callsite... Carry on :-).